Add new Traffic management concept. (#4321)
istio/istio.io: content/docs/concepts/traffic-management/routing/gateways/index.md

---
title: Gateways
description: Learn about using gateways to configure how the Envoy proxies load balance HTTP, TCP, or gRPC traffic.
weight: 3
keywords: [traffic-management, http, tcp, grpc, ingress, egress]
---

You use a [gateway](/docs/reference/config/networking/v1alpha3/gateway/)
to manage inbound and outbound traffic for your mesh. You can manage the

> following types of traffic with a gateway: HTTP TCP gRPC

Reviewer: Why are you listing these 3 protocols?
https://preliminary.istio.io/docs/reference/config/networking/v1alpha3/gateway/#Port

Author: Because I didn't remember we had all ports listed there. Thanks.

> A gateway configurations apply to Envoy proxies that are separate from the

Reviewer: ```suggestion
Gateway configurations apply to Envoy proxies that are separate from the
```

service sidecar proxies. To configure a gateway means configuring an Envoy
proxy to allow or block certain traffic from entering or leaving the mesh.

Your mesh can have any number of gateway configurations, and multiple gateway
workload implementations can co-exist within your mesh. You might use multiple
gateways to have one gateway for private traffic and another for public

> traffic, so that you can balance the loads independently.

Reviewer: ```suggestion
traffic, so you can keep all private traffic inside a firewall, for example.
```

    tls:
      mode: SIMPLE
      serverCertificate: /tmp/tls.crt
      privateKey: /tmp/tls.key
{{< /text >}}

This gateway configuration lets HTTPS traffic from `ext-host` into the mesh on
port 443, but doesn't specify any routing for the traffic.

### Bind a gateway to a virtual service

To specify routing and for the gateway to work as intended, you must also bind
the gateway to a virtual service. You do this using the virtual service's
`gateways:` field, as shown in the following example:

{{< text yaml >}}
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: virtual-svc
spec:
  hosts:
    - ext-svc
  gateways:
    - ext-host-gwy
{{< /text >}}

You can then configure the virtual service with routing rules for the external
traffic. The following diagram shows the configured routes for the external
HTTPS traffic to the `my-svc` service within the mesh:

{{< image width="70%"
    link="./gateways-2.svg"
    caption="Configurable routes for the external HTTPS traffic"
    >}}

For more information:

-  Refer to the [gateways reference documentation](/docs/reference/config/networking/v1alpha3/gateway/)
   to review all the enabled keys and values.

> -  Refer to the Ingress task topic for instructions, including how to configure

Reviewer: ```suggestion
-  Refer to the [Ingress task topic](/docs/tasks/traffic-management/ingress/) for instructions, including how to configure
```

   an Istio gateway for Kubernetes ingress.

> -  Refer to the Egress task topic to learn how to configure egress traffic

Reviewer: ```suggestion
-  Refer to the [Egress task topic](/docs/tasks/traffic-management/egress/) to learn how to configure egress traffic
```

---
title: Gateways
description: Learn about using gateways to configure how the Envoy proxies load balance HTTP, TCP, or gRPC traffic.
weight: 3
keywords: [traffic-management, http, tcp, grpc, ingress, egress]
---

You use a [gateway](/docs/reference/config/networking/v1alpha3/gateway/) to
manage inbound and outbound traffic for your mesh. You can manage
[multiple types of traffic](/docs/reference/config/networking/v1alpha3/gateway/#Port)
with a gateway.

Gateway configurations apply to Envoy proxies that are separate from the
service sidecar proxies. To configure a gateway means configuring an Envoy
proxy to allow or block certain traffic from entering or leaving the mesh.

Your mesh can have any number of gateway configurations, and multiple gateway
workload implementations can co-exist within your mesh. You might use multiple
gateways to have one gateway for private traffic and another for public
traffic, so you can keep all private traffic inside a firewall, for example.

> For better security, you can use a gateway to make your services inaccessible

Reviewer 2: This sentence is not right. Please remove.
